clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainStorageEngine.java

```diff
@@ -47,35 +48,64 @@
   private static final int CHAIN_HEADER_TAIL_OFFSET = 8;
   private static final int CHAIN_HEADER_SIZE = 16;
 
+  private static final int DETACHED_CONTIGUOUS_CHAIN_XTN_ADDRESS_OFFSET = 0;
+  private static final int DETACHED_CONTIGUOUS_CHAIN_HEADER_SIZE = 8;
+
   private final OffHeapStorageArea storage;
   private final Portability<? super K> keyPortability;
   private final Set<AttachedInternalChain> activeChains = Collections.newSetFromMap(new ConcurrentHashMap<AttachedInternalChain, Boolean>());
+  private final int extendedChainHeaderSize;
+  private final ByteBuffer emptyExtendedChainHeader;
+  private final int totalChainHeaderSize;
+  private final OffHeapChainExtensionHeader extendedHeader;
 
   private StorageEngine.Owner owner;
   private long nextSequenceNumber = 0;
```

Comment:
Actually everything in the storage engine assumes appropriate locks are held on the map and since writeMapping etc.. are all under write lock we should be fine here. besides this is not a code associated with the current change.